op/model.py

```diff
@@ -174,7 +174,7 @@ def __init__(self, relation_name, relation_id, our_unit, backend, cache):
                 self.units.add(unit)
                 if self.app is None:
                     self.app = unit.app
-        except RelationNotFound:
```

Comment:
I suggest just dropping this error for now, and removing that class, so this bubbles up as a ModelError as well if it happens. This way we avoid the hack of trying to guess what the error message is for now.

Reply:
Per our discussion, we cannot just drop it because we need to handle loading of relation data for dead relations when handling deferred events. In other words, we cannot return an empty dictionary based on **any** ModelError.

```
       except RelationNotFoundError:
            # Dead relations tell no tales (and have no data).
            return {}
```